lib/puppet/provider/service/systemd.rb

```diff
@@ -123,6 +123,15 @@ def debian_enabled?
     end
   end
 
+  def daemon_reload?
```

Comment:
Define the daemon_reload? function to check if the unit is requiring to trigger a "systemctl daemon-reload"
This function is called only on start & restart unit options.